src/components/MainView.js

```diff
@@ -1,13 +1,39 @@
 import React, { useCallback, useEffect, useState } from 'react'
+import { useHistory, useLocation } from 'react-router-dom'
 import { Layout, Root, ScrollView, useViewport } from '@aragon/ui'
 import MenuPanel, { MENU_PANEL_WIDTH } from './MenuPanel'
 import Header from './Header/Header'
+import { getPreferencesSearch } from '../Routes'
+import GlobalPreferences from './GlobalPreferences/GlobalPreferences'
+
+const GLOBAL_PREFERENCES_QUERY_PARAM = '?preferences=/'
 
 function MainView({ children }) {
   const { width: vw, below } = useViewport()
   const compactMode = below('medium')
-
   const [menuPanelOpen, setMenuPanelOpen] = useState(!compactMode)
+  const history = useHistory()
+  const search = useLocation().search
+  const locator = useLocation()
+
+  function parsePreferences(search = '') {
```

Comment:
As mentioned earlier, there is likely a better way to handle query parameters like this within react-router.

Reply:
did a research and asked Pierre about this and didn't find a better way to handle the parameters, seems that we need to handle the query parameters by ourselves